VersionBlock.js

```diff
@@ -1,108 +1,207 @@
-import React, { useContext, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import {
   View,
   Text,
   TouchableOpacity,
+  Image,
 } from 'react-native';
 import {
-  SimpleHTMLView,
-  SText,
+    openActionSheet,
+    SimpleHTMLView,
+    SText,
 } from './Misc';
-import { GlobalStateContext, getTheme } from './StateManager';
 import styles from './Styles.js';
+import {useGlobalState} from "./Hooks";
+import {VersionFilter} from "./Filter";
+import {useState} from "react";
+import {iconData} from "./IconData";
+import strings from './LocalizedStrings';
 
 
+const useVersionTitle = (version) => {
+  const { interfaceLanguage } = useGlobalState();
+  if (version.merged) {
+    return strings.mergedFrom + " " + Array.from(new Set(version.sources)).join(", ");
+  } else if (interfaceLanguage === "english" || !version.versionTitleInHebrew) {
+    return version.versionTitle;
+  } else {
+    return version.versionTitleInHebrew;
+  }
+}
+
+const useTextAlign = () => {
+  const {interfaceLanguage} = useGlobalState();
+  return {textAlign: (interfaceLanguage === 'hebrew') ? 'right' : 'left'};
+}
+
 const VersionBlock = ({
   version,
-  openVersionInSidebar,
-  openVersionInReader,
   openUri,
-  handleOpenURL,
+  openFilter,
+  segmentRef
 }) => {
-  const { themeStr, textLanguage, interfaceLanguage } = useContext(GlobalStateContext);
-  const theme = getTheme(themeStr);
-
-  const onVersionTitleClick = useCallback(() => {
-    const action = openVersionInSidebar ? openVersionInSidebar : openVersionInReader;
-    if (action) {
-      action(version.versionTitle, version.versionTitleInHebrew, version.language);
-    }
-  }, [version]);
-  const onSelectVersionClick = useCallback(() => {
-    if (openVersionInReader) {
-      openVersionInReader(version.versionTitle, version.language);
-    }
-  }, [version]);
-
-  let versionTitle, versionSource, shortVersionSource, license, licenseURL, versionNotes;
-  if (Sefaria.util.get_menu_language(interfaceLanguage, textLanguage) == "hebrew") {
-    versionTitle = version['versionTitleInHebrew'] || version['versionTitle'];
-    versionSource = version['versionSource'];
-    shortVersionSource = Sefaria.util.parseURLhost(versionSource);
-    license = version['license'];
-    licenseURL = Sefaria.util.getLicenseURL(license);
-    versionNotes = version['versionNotesInHebrew'] || version['versionNotes'];
-  } else {
-    versionTitle = version['versionTitle'];
-    versionSource = version['versionSource'];
-    shortVersionSource = Sefaria.util.parseURLhost(versionSource);
-    license = version['license'];
-    licenseURL = Sefaria.util.getLicenseURL(license);
-    versionNotes = version['versionNotes'];
-  }
+  const {theme} = useGlobalState();
+  const innerText = (
+    <SText lang={"english"} style={[styles.en, styles.textTocVersionTitle, useTextAlign(), theme.text]}>
```

Comment:
it's dangerous to put a hook as a style because there may be a conditional added before this line that returns early. Better to define it at the top of the component.

Reply:
Or change the function to be `getTextAlign(interfaceLanguage)` so it's not a hook